core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -481,6 +481,23 @@ static HttpResponse of(ResponseHeaders headers, Publisher<? extends HttpObject>
         return PublisherBasedHttpResponse.from(headers, publisher);
     }
 
+    /**
+     * Creates a new HTTP response with the specified headers and trailers
+     * whose stream is produced from an existing {@link Publisher}.
+     *
+     * <p>Note that the {@link HttpObject}s in the {@link Publisher} are not released when
+     * {@link Subscription#cancel()} or {@link #abort()} is called. You should add a hook in order to
+     * release the elements. See {@link PublisherBasedStreamMessage} for more information.
+     */
+    static HttpResponse of(ResponseHeaders headers,
```

Comment:
Could you also add a method that uses `AppendingPublisher `to `HttpRequest`?

Reply:
https://github.com/line/armeria/blob/b16bc8d926e3b4ebb07b86b66a09cd1126121574/core/src/main/java/com/linecorp/armeria/common/HttpRequest.java#L270-L281

On `HttpRequest `, it already used `PublisherBasedHttpRequest` 🤔 
What method can I apply `SurroundingPublisher` on `HttpRequset`?

Reply:
We may add the same variant to `HttpRequest` although trailers are rarely used for requests but possible. 
```java
static HttpRequest of(RequestHeaders headers, Publisher<? extends HttpData> publisher, HttpHeaders trailers) { ... }
```

Reply:
✅ updated! Add `HttpRequest.of(headers, publisher, trailers)` using `SurroundingPublisher`